packages/rendermime-interfaces/src/index.ts

```diff
@@ -227,6 +228,20 @@ export namespace IRenderMime {
     readonly default: IExtension | ReadonlyArray<IExtension>;
   }
 
+  /**
+   * The IJLIcon interface, which supplies element, render,
+   * and unrender functions
+   */
+  export interface IJLIcon extends VirtualElementPass.IRenderer {
+    /**
+     * Create an icon as a DOM element
+     *
+     * @returns A DOM element that contains an (inline) svg element
```

Comment:
Does it *have* to be an inline svg element? Or might that change in the future, and that is just an implementation detail? If it is just an implementation detail, it should not be in the interface documentation.

This is an interface we do not want to break, because we want to basically never have a backwards incompatible change in rendermime-interfaces. So we should be as generic as we can here and still obtain our objective.

Reply:
> because we want to basically never have a backwards incompatible change in rendermime-interfaces

Ah. I was not aware of that detail. In that case, we should probably do away with the `IJLIcon.element` function entirely. Basically, `element` is in that interface in order to account for the fact that the file type icons are handled differently in the tabbars and in the filebrowser listing. If I fix things up so that `render` and `unrender` can be used to create the icons in both contexts, that eliminates the need for `element` here.

Reply:
I've now removed `IJLIcon.element`